app/assets/javascripts/doughnut_chart.js

```diff
@@ -0,0 +1,77 @@
+function getChartColor(amount) {
+  if(amount < 80) {
+    return 'green';
+  }
+  if(amount < 90) {
+    return 'yellow';
+  }
+  return 'red';
+}
+
+function getChartOptions(amount, title) {
+  var color = getChartColor(amount);
+  var backgroundColors = {
+    green: 'rgba(0, 255, 0, 0.3)',
+    yellow: 'rgba(255, 255, 0, 0.3)',
+    red: 'rgba(255, 0, 0, 0.3)',
+  };
+  var borderColors = {
+    green: 'rgba(0, 255, 0, 0.5)',
+    yellow: 'rgba(255, 255, 0, 0.5)',
+    red: 'rgba(255, 0, 0, 0.5)',
+  };
+
+  return {
+      type: 'doughnut',
+      data: {
+        datasets: [{
+          data: [amount, 100 - amount],
+          backgroundColor: [
+            backgroundColors[color],
+            'rgba(255, 255, 255, 0.5)',
+          ],
+          borderColor: [
+            borderColors[color],
+            'rgba(0, 0, 0, 0.2)'
+          ],
+          borderWidth: 1
+        }]
+      },
+      options: {
+        cutoutPercentage: 85,
+        rotation: (3/4) * Math.PI,
+        circumference: (3/2) * Math.PI,
+        tooltips: {
+          enabled: false
+        },
+        title: {
+          display: true,
+          padding: -6,
+          fontSize: 24,
+          text: title,
+          position: 'bottom'
+        },
+        events: []
+      }
+    }
+}
+
+function makeHalfDoughnutChart(name, amount, description) {
+    var canvas = document.getElementById(name);
+    canvas.style.width = "150px";
+    canvas.style.height = "150px";
```

Comment:
Can you maybe make the width and height relative and not absolute? In my mind this would then mean that when resizing the window, the canvas size would scale accordingly. (I have no idea if this is possible though :D)

Reply:
a relative amount would be relative to viewport width. Long story short: a small window size makes the charts look tiny. Furthermore, a fix px assignment does not really mean what it sounds like: It is more a hypothetical px when rendered in a browser (px does not mean a real pixel on screen).

Reply:
Ah, thanks for the clarification! :+1: